presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftServiceClient.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.swift.service.ThriftException;
+import com.facebook.swift.service.ThriftMethod;
+import com.facebook.swift.service.ThriftService;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Presto Thrift service definition.
+ * This thrift service needs to be implemented on the Thrift Node in order to be used with Thrift Connector.
+ */
+@ThriftService("presto-thrift")
+public interface ThriftServiceClient
```

Comment:
One reason is that it's difficult to cache results of such methods which is critical for metadata ones. Also, I assumed that results of metadata methods should be deterministic and not depend on session values.